ha_lineairdb.cc

```diff
@@ -199,14 +201,22 @@ static int lineairdb_init_func(void* p) {
   return 0;
 }
 
+static std::shared_ptr<LineairDB::Database> get_or_allocate_database(LineairDB::Config conf) {
```

Comment:
I think this function isn't thread-safe. You must use mutex (mutual exclusion locking) to prevent creating `LineairDB::Database` twice in concurrent.

Reply:
In addition you can get better concurrency by using "double-check locking" strategy.
FYI: https://en.wikipedia.org/wiki/Singleton_pattern#Lazy_initialization

Reply:
I used the std::call_once and std::once_flag implementation in the latest commit d5bba61
https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_C++11

Reply:
> I used the std::call_once and std::once_flag implementation

Good! I didn't know this helper. Nice to know about it.